src/components/UserInput/UserInput.js

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+import './UserInput.scss';
+
+const UserInput = ({ name, koreanName, condition, loginInputHandler }) => {
+  const [loginValidationBtn, setLoginValidationBtn] = useState(false);
+
+  const validationChecker = condition => {
+    if (condition) {
+      setLoginValidationBtn(true);
+    } else {
+      setLoginValidationBtn(false);
+    }
+  };
+
+  const [inputTitleRedBtn, setInputTitleRedBtn] = useState(false);
+  const [inputTitleGoUpBtn, setInputTitleGoUpBtn] = useState(false);
+
+  const inputFocusEffector = e => {
+    setInputTitleGoUpBtn(true);
+  };
+
+  const inputBlurEffector = e => {
+    e.target.value ? setInputTitleGoUpBtn(true) : setInputTitleGoUpBtn(false);
+  };
+
+  const inputRedEffector = () => {
+    loginValidationBtn ? setInputTitleRedBtn(false) : setInputTitleRedBtn(true);
+  };
+  return (
+    <>
+      <div className="user-input">
+        <span
+          className={
+            'user-placeholder ' +
+            (inputTitleGoUpBtn ? 'input-title-goup' : null)
+          }
+          style={inputTitleRedBtn ? { color: 'red' } : { color: 'gray' }}
+        >
+          {koreanName} *
+        </span>
```

Comment:
```suggestion
         <span
          className={`user-placeholder 
            ${inputTitleGoUpBtn ? 'input-title-goup' : null}
          ${!condition ? 'input-red' : 'input-gray'}`}
        >
          {koreanName} *
        </span>
```
해당 부분은 위와 같이 classname을 처리할 수 있긴 한데, 복잡해 보이네요
조금 더 좋은 방법을 생각해 보는 것이 필요할 것 같습니다